subprojects/gradle/android/src/main/kotlin/com/avito/android/AndroidSdk.kt

```diff
@@ -3,92 +3,73 @@ package com.avito.android
 import com.avito.utils.ExistingDirectory
 import com.avito.utils.ProcessRunner
 import org.gradle.api.Project
+import org.gradle.api.logging.Logger
 import java.io.File
 import java.util.Properties
 
 val Project.androidSdk: AndroidSdk
-    get() = AndroidSdk(this, ProcessRunner.Real())
+    get() = AndroidSdk.fromAndroidProject(this)
 
-class AndroidSdk(
-    private val project: Project,
-    private val processRunner: ProcessRunner
+open class BaseAndroidSdk(
```

Comment:
Needed to separate two use cases: 
- SDK model based on Android module. Here we have BaseExtension with exact versions.
- Common Android SDK model. I've decided not to speculate here about any versions.